features/signals_railway_signals.yaml

```diff
@@ -963,6 +963,20 @@ signals_railway_signals:
         - { tag: 'railway:signal:main', values: ['DE-AVG:f', 'DE-BOStrab:f'] }
         - { tag: 'railway:signal:main:form', value: 'light' }
 
+    - description: U-Bahn Nürnberg main signal
+      country: DE
+      icon:
+        match: 'railway:signal:main:states'
+        cases:
+          - { regex: '^(.*;)?DE-VAGN:hp2(;.*)?$', value: 'de/vag-nuremberg/hp2' }
+          - { regex: '^(.*;)?DE-VAGN:hp1(;.*)?$', value: 'de/vag-nuremberg/hp1' }
+          - { regex: '^(.*;)?off(;.*)?$', value: 'de/vag-nuremberg/off' }
+          - { regex: '^(.*;)?DE-VAGN:hp3(;.*)?$', value: 'de/vag-nuremberg/hp3' }
+        default: 'de/vag-nuremberg/hp0'
+      tags:
+        - { tag: 'railway:signal:main', value: 'DE-BOStrab:h' }
```

Comment:
Looking at https://taginfo.openstreetmap.org/keys/railway%3Asignal%3Amain%3Astates#values, and then searching in the table for "VAGN" I get all signals which are mapped with `railway:signal:main=DE-BOStrab:h`.

![image](https://github.com/user-attachments/assets/8488750f-ae4a-4744-979a-c4167549576c)

So I think we can merge this section with the existing section below for `railway:signal:main=DE-BOStrab:h`, and then make cases for each variant of that signal (so in the future also e.g. Stuttgart, Hamburg).

Reply:
> This one matches on the same tags as "tram Hauptsignal" below. That one can probably be removed
> 

I'm unsure what you mean, I did it this way because it's analogue to Hamburger Hochbahn (HHA) signals, which are sort of their own system, but actually fall into the category BOStrab:h

> Also "tram Hauptsignal mit Vorsignal" matches on the same tags, but for `railway:signal:combined` instead of `railway:signal:main`. Should icons also be added / updated for the combined tram signals?

That's a tagging mistake, Nuremberg only has main and distant subway signals.

Reply:
> Looking at https://taginfo.openstreetmap.org/keys/railway%3Asignal%3Amain%3Astates#values, and then searching in the table for "VAGN" I get all signals which are mapped with `railway:signal:main=DE-BOStrab:h`.
> 
> ![image](https://private-user-images.githubusercontent.com/1073881/395810132-8488750f-ae4a-4744-979a-c4167549576c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQyNzcxMTUsIm5iZiI6MTczNDI3NjgxNSwicGF0aCI6Ii8xMDczODgxLzM5NTgxMDEzMi04NDg4NzUwZi1hZTRhLTQ3NDQtOTc5YS1jNDE2NzU0OTU3NmMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTIxNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEyMTVUMTUzMzM1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NWQxNmNmNjBmMmE2MDllY2JiZWJkZmU2MjhkNjBlMTYzY2Q3NWI2MDM2NzRiOWMxMTg3YWM3NGYzZDk5NDIwYSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.UdgL0MfqdFm7Nk53zuT6r8J4_Hv7DTP1MsIHyS1Oju0)
> 
> So I think we can merge this section with the existing section below for `railway:signal:main=DE-BOStrab:h`, and then make cases for each variant of that signal (so in the future also e.g. Stuttgart, Hamburg).

I see, will do that.

Reply:
Eh I just realized, for the off state I'd have to retag all these signals with something else than BOStab:h... :(